lib/ruleHelper.js

```diff
@@ -93,6 +96,98 @@ RuleHelper.prototype = {
         return allowed;
     },
 
+    /**
+     * Check if an identifier is allowed
+     * - find its declarations and see if it's const or let
+     * - if so, allow if the declaring statement is an allowed expression
+     * - ensure that following assignments to that identifier are also allowed
+     *
+     * @param {Object} expression Identifier expression
+     * @param {Object} escapeObject contains keys "methods" and "taggedTemplates" which are arrays of strings
+     *                              of matching escaping function names.
+     * @param {Object} details Additional linter violation state information, in case this function was called
+     *                         recursively.
+     * @returns {boolean} Returns whether the Identifier is deemed safe.
+     */
+    isAllowedIdentifier(expression, escapeObject, details) {
+        // find declared variables and see which are literals
+        const scope = this.context.getScope(expression);
+        const variableInfo = scope.set.get(expression.name);
+        let allowed = false;
+
+        // If we can't get info on the variable, we just can't allow it
+        if (!variableInfo ||
+            !variableInfo.defs ||
+            !variableInfo.defs.length > 0 ||
+            !variableInfo.references ||
+            !variableInfo.references.length > 0) {
+            // FIXME Fix/Adjust towards a helpful message here and update tests accordingly.
+            // details.message = `Variable ${expression.name} considered unsafe: variable initialization not found`;
+            return false;
+        }
+
+        // look if the var was defined as allowable
+        let definedAsAllowed = false;
+        for (const def of variableInfo.defs) {
+            if (def.node.type !== "VariableDeclarator") {
+                // What else could it be..
+                const {line, column} = def.node;
+                details.message = `Variable '${expression.name}' initialized with unknown declaration at ${line}:${column}`;
+
```

Comment:
I haven't thought which js code fragment may actually be able to trigger it, but I think that here we should also set `definedAsAllowed = false` explicitly, in case there was another `variableInfo.defs` entry before that made us to it the `definedAsAllowed = true` at line 159.

Reply:
Added it to line 136 with my upcoming commit.